File: cpp/q007.cpp
#include<iostream>
#include<cmath>
#include<string>
#include "euler.hpp"

using namespace std;



int main(){
    int count=0;
    int num=2;
    while (count<10001){
        if (PrimeUtils::isPrime(num)){
            count+=1;
        }
        num+=1;
    }
    cout << num-1 <<endl;
    return 0;
}

File: rust/q007.rs
fn is_prime(n: u32) ->bool{
    let lim = (n as f32).sqrt() as u32;
    if n==2{
        return true;
    }
    if n%2==0{
        return false;
    }
    for i in 3..lim+1{
        if n%i==0{
            return false;
        }
    }
    return true;
}

fn main(){
    let mut count=0;
    let mut num=2;
    while count<10001{
        if is_prime(num){
            count+=1;
        }
        num+=1;
    }
    println!("{}", num-1)
}